app/lib/actions/katello/repository/filtered_index_content.rb

```diff
@@ -21,16 +22,25 @@ def run
             ::Katello::FileUnit.import_for_repository(repo)
           elsif repo.deb?
             ::Katello::Deb.import_all(unit_ids, repo)
+          elsif repo.yum?
+            if input[:content_type] == 'srpm'
+              ::Katello::Srpm.import_all(unit_ids, repo)
           else
             ::Katello::Rpm.import_all(unit_ids, repo)
+            end
           end
         end
 
         private
 
         def search_units(repo)
-          found = repo.unit_search(:type_ids => [repo.unit_type_id],
+          if input[:content_type] == 'srpm'
+            found = repo.unit_search(:type_ids => input[:content_type],
+                                     :filters => input[:filter])
+          else
+            found = repo.unit_search(:type_ids => [repo.unit_type_id],
```

Comment:
instead of 'defaulting' to  repo.unit_type_id, use default_managed_content_type in the repository manager.